File: manual/algorithm/排序/插入排序/insertSort.js
#!/usr/bin/env node

const insertSort = (a) => {
  if (!a) return;
  for (let i = 1; i < a.length; i++) {
    let key = a[i], j = i - 1;
    for (; j >= 0 && a[j] > key; j--) {
      a[j + 1] = a[j];
    }
    a[j + 1] = key;
  }
}

console.time('insertSort');
let arr = [];
for (let i = 0; i < 100; i++)
  arr.push(Math.round(Math.random() * 10));
insertSort(arr);
// console.log(arr);
console.timeEnd('insertSort');

File: manual/algorithm/排序/插入排序/insertSort.py
from random import random
import time

def insertSort(arr):
  if arr:
    for i in range(1, len(arr)):
      key, j = arr[i], i - 1
      while j >= 0 and arr[j] > key:
        arr[j + 1] = arr[j]
        j -= 1
      arr[j + 1] = key
  
  return

start = time.time()
arr = list()

for i in range(0, 100):
  arr.append(round(random() * 10))

insertSort(arr)
# print(arr)
delta = time.time() - start
print(f"time={delta:.3f}ms")